File: backend/voting-app/src/main/java/com/tu/votingapp/utils/mappers/PermissionMapper.java
package com.tu.votingapp.utils.mappers;

import com.tu.votingapp.dto.general.PermissionDTO;
import com.tu.votingapp.entities.PermissionEntity;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface PermissionMapper {
    PermissionDTO toDto(PermissionEntity entity);
    PermissionEntity toEntity(PermissionDTO dto);
}

File: frontend/android/app/src/main/java/com/example/android/utils/mappers/PermissionMapper.kt
package com.example.android.utils.mappers

import com.example.android.dto.response.PermissionDTO
import com.example.android.entity.PermissionEntity

object PermissionMapper {

    fun PermissionDTO.toEntity(): PermissionEntity = PermissionEntity(
        id = this.id,
        name = this.name
    )

    fun PermissionEntity.toDto(): PermissionDTO = PermissionDTO(
        id = this.id!!,
        name = this.name
    )
}